app/controllers/application_controller/filter.rb

```diff
@@ -956,7 +959,11 @@ def adv_search_build(model)
         s = MiqSearch.find(@edit[:expression][:selected][:id])
         clear_default_search if s.search_key == "_hidden_"
       end
-      @edit.delete(:exp_token)                                          # Remove any existing atom being edited
+      unless @edit.nil?
```

Comment:
@lgalis I stubled upon this and tried to fix it because I thought that those were errors related to #1441 What do you think? Are you ok with this fix, or should I revert these changes and just fix problems which are directly related to #1441?